c7n/resources/servicecatalog.py

```diff
@@ -0,0 +1,170 @@
+from c7n.query import ConfigSource, DescribeSource, QueryResourceManager, TypeInfo
+from c7n.manager import resources
+from c7n.exceptions import PolicyValidationError
+from c7n.tags import universal_augment
+from c7n.utils import local_session, type_schema
+
+
+class DescribePortfolio(DescribeSource):
+
+    def augment(self, resources):
+        return universal_augment(self.manager, super().augment(resources))
+
+
+@resources.register('catalog-portfolio')
+class CatalogPortfolio(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'servicecatalog'
+        enum_spec = ('list_portfolios', 'PortfolioDetails', None)
+        detail_spec = ('describe_portfolio', 'Id', 'Id', None)
+        arn = 'ARN'
+        id = 'Id'
+        name = 'DisplayName'
+        date = 'CreatedTime'
+        universal_taggable = object()
+        cfn_type = config_type = 'AWS::ServiceCatalog::Portfolio'
+
+    source_mapping = {
+        'describe': DescribePortfolio,
+        'config': ConfigSource
+    }
+
+
+@CatalogPortfolio.action_registry.register('delete')
+class CatalogPortfolioDeleteAction(BaseAction):
+    """Action to delete a Service Catalog Portfolio
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: service-catalog-portfolio-delete
+            resource: aws.catalog-portfolio
+            filters:
+              - type: cross-account
+            actions:
+              - delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ('servicecatalog:DeletePortfolio',)
+
+    def process(self, portfolios):
+        client = local_session(self.manager.session_factory).client('servicecatalog')
+        for r in portfolios:
+            self.manager.retry(client.delete_portfolio, Id=r['Id'], ignore_err_codes=(
+                'ResourceNotFoundException',))
+
+
+@CatalogPortfolio.filter_registry.register('cross-account')
+class CatalogPortfolioCrossAccount(CrossAccountAccessFilter):
+    """Check for account ids that the service catalog portfolio is shared with
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: catalog-portfolio-cross-account
+                resource: aws.catalog-portfolio
+                filters:
+                  - type: cross-account
+    """
+
+    schema = type_schema(
+        'cross-account',
+        whitelist_accounts_from={'$ref': '#/definitions/filters_common/value_from'},
+        whitelist={'type': 'array', 'items': {'type': 'string'}})
+
+    permissions = ('servicecatalog:ListPortfolioAccess',)
+    annotation_key = 'c7n:CrossAccountViolations'
+
+    def violation_checker(self, client, accounts, resources):
+        results = []
+        shared_accounts = set()
+        for r in resources:
+            accounts = self.manager.retry(
+                client.list_portfolio_access, PortfolioId=r['Id'], ignore_err_codes=(
+                    'ResourceNotFoundException',))
+            if accounts:
```

Comment:
this should if not accounts: continue

else shared_accounts won't be defined and the reference to it will be an error later on

Reply:
ah, true! thanks for the quick fix